server/src/main/java/io/druid/client/DirectDruidClient.java

```diff
@@ -168,16 +176,24 @@ public int getNumOpenConnections()
       final QueryMetrics<? super Query<T>> queryMetrics = toolChest.makeMetrics(query);
       queryMetrics.server(host);
 
+      long timeoutAt = ((Long) context.get(QUERY_FAIL_TIME)).longValue();
```

Comment:
QueryResource is neither official API in #4433. Making a query as HTTP request to the same JVM runtime is ridiculous.